src/main/java/org/opensearch/knn/index/codec/util/KNNCodecUtil.java

```diff
@@ -94,24 +94,25 @@ public static KNNCodecUtil.Pair getFloats(BinaryDocValues values) throws IOExcep
     }
 
     public static long calculateArraySize(int numVectors, int vectorLength, SerializationMode serializationMode) {
+        // For more details on array memory usage in Java, visit https://www.javamex.com/tutorials/memory/array_memory_usage.shtml
         if (serializationMode == SerializationMode.ARRAY) {
             int vectorSize = vectorLength * FLOAT_BYTE_SIZE + JAVA_ARRAY_HEADER_SIZE;
             if (vectorSize % JAVA_ROUNDING_NUMBER != 0) {
-                vectorSize += vectorSize % JAVA_ROUNDING_NUMBER;
+                vectorSize += (JAVA_ROUNDING_NUMBER - vectorSize % JAVA_ROUNDING_NUMBER);
             }
             int vectorsSize = numVectors * (vectorSize + JAVA_REFERENCE_SIZE) + JAVA_ARRAY_HEADER_SIZE;
             if (vectorsSize % JAVA_ROUNDING_NUMBER != 0) {
-                vectorsSize += vectorsSize % JAVA_ROUNDING_NUMBER;
+                vectorsSize += (JAVA_ROUNDING_NUMBER - vectorsSize % JAVA_ROUNDING_NUMBER);
             }
             return vectorsSize;
         } else {
```

Comment:
Could you provide a reference for this case?

Reply:
It's the same link as above, under the 2-D array section

Reply:
2-D array is for the first one. This one is not 2-D array I believe?

Reply:
The first one is a 1D array since it's an individual vector.  The 2D is an array of vectors